pkg/action/sysext.go

```diff
@@ -0,0 +1,384 @@
+// them into /run/extensions if the signature doesnt match, so the measurements are not that important currently and
+// maybe they are not useful if sysext is the way of upgrading your system
+
+// So potentially we have a unified system where we have the same behaviour for both, but that measn that
+// anything provided by sysext will NOT be available until the initramfs stage where we enable the sysext service
+// we can potentially move the sysext service as early as possible once we mount the persistent dir
+
+// Extensions are provided in the following formats:
+// - .raw : image based extension
+// - any directory : directory based extension
+
+// TODO: Check which extensions are running
+// TODO: On disable we should check if the extension is running and refresh systemd-sysext?
+// TODO: On enable, should we refresh systemd-sysext immediately? Only if the current boot state equals the bootstate we enabled it for?
+
+const (
+	sysextDir        = "/var/lib/kairos/extensions/"
+	sysextDirActive  = "/var/lib/kairos/extensions/active"
+	sysextDirPassive = "/var/lib/kairos/extensions/passive"
+)
+
+// SysExtension represents a system extension
+type SysExtension struct {
+	Name     string
+	Location string
+}
+
+func (s *SysExtension) String() string {
+	return s.Name
+}
+
+// ListSystemExtensions lists the system extensions in the given directory
+// If none is passed then it shows the generic ones
+func ListSystemExtensions(cfg *config.Config, bootState string) ([]SysExtension, error) {
+	switch bootState {
+	case "active":
+		cfg.Logger.Debug("Listing active system extensions")
+		return getDirExtensions(cfg, sysextDirActive)
+	case "passive":
+		cfg.Logger.Debug("Listing passive system extensions")
+		return getDirExtensions(cfg, sysextDirPassive)
+	default:
+		cfg.Logger.Debug("Listing all system extensions (Enabled or not)")
+		return getDirExtensions(cfg, sysextDir)
+	}
+}
+
+// getDirExtensions lists the system extensions in the given directory
+func getDirExtensions(cfg *config.Config, dir string) ([]SysExtension, error) {
+	var out []SysExtension
+	// get all the extensions in the sysextDir
+	entries, err := cfg.Fs.ReadDir(dir)
+	// We don't care if the dir does not exist, we just return an empty list
+	if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+	for _, entry := range entries {
+		if !entry.IsDir() && filepath.Ext(entry.Name()) == ".raw" {
+			out = append(out, SysExtension{Name: entry.Name(), Location: filepath.Join(dir, entry.Name())})
+		}
+	}
+	return out, nil
+}
+
+// GetSystemExtension returns the system extension for a given name
+func GetSystemExtension(cfg *config.Config, name, bootState string) (SysExtension, error) {
+	// Get a list of all installed system extensions
+	installed, err := ListSystemExtensions(cfg, bootState)
+	if err != nil {
+		return SysExtension{}, err
+	}
+	// Check if the extension is installed
+	// regex against the name
+	re, err := regexp.Compile(name)
```

Comment:
[minor] we may need to constrain this a bit more to avoid things like "fips" matching "non-fips" or similar.

Reply:
well its a regex no? So doing `^fips` should not match `non-fips`